components/automate-ui/src/app/pages/project/list/project-list.component.ts

```diff
@@ -82,8 +85,21 @@ export class ProjectListComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.store.dispatch(new GetProjects());
+    // Get status now; periodic checks are already being done globally so no need to poll further
     this.projects.getApplyRulesStatus();
+
+    // Get projects status now and periodically while on this page
+    this.store.dispatch(new GetProjects());
+    observableInterval(1000 * this.POLLING_INTERVAL_IN_SECONDS).pipe(
```

Comment:
Not sure about the UX of this polling. I see the project list blip every 10 seconds, but it doesn't react at the same time that the `Update Button` becomes re-enabled. Instead of polling why don't we call `GetProjects()` as soon as the update is completed? 

This might require rethinking our current `apply-rules` polling. It'd be great if we only started polling when a project update was in progress.

Reply:
A few things here. 
1. In a typical scenario, the projects list will go from "edits pending" to "applied" on the first poll, because that strictly reflects the DB update that we do almost immediately. The updated button, on the other hand, will be otherwise occupied for hours. So your observation that they don't occur simultaneously will almost always be true. Also, it would be very late, then, to wait until the update is complete before calling `GetProjects`.
2. When the update is complete will have a direct affect on the Project Update Status, not the Ingest Rules column, and that will be in the next PR.
3. Regarding polling, it is necessary not for the actions of this user but because of potential actions of other users. Whenever we are on the projects list page, it behooves us to know if someone else has queued some project changes and we will then see those posted every 10 seconds. That is also quite independent of whether we have started an update or not, so the polling needs to be active when simply on this page, unconditionally.